packages/gatsby-cli/src/reporter/reporter.ts

```diff
@@ -36,6 +41,42 @@ class Reporter {
   stripIndent = stripIndent
   format = chalk
 
+  errorMap: Record<ErrorId, IErrorMapEntry> = {}
+
+  /**
+   * Set a custom error map to the reporter. This allows
+   * the reporter to extend the internal error map
+   */
+
+  setErrorMap = (entry: Record<ErrorId, IErrorMapEntry>): void => {
+    this.errorMap = {
+      ...this.errorMap,
+      ...entry,
+    }
+  }
+
+  setErrorMapWithPluginName = (name: string) => (
+    entry: Record<ErrorId, IErrorMapEntry>
+  ): void => {
+    const entries = Object.entries(entry)
+
+    const newErrorMap = entries.reduce((memo, [key, val]) => {
+      memo[`${name}_${key}`] = val
+
+      return memo
+    }, {})
+
+    this.setErrorMap(newErrorMap)
+  }
```

Comment:
do we want to expose it on the reporter? Maybe it's better to export a utility that we use in api-runner. It feels like a private function